Tests/ImperialTests/ImperialTests.swift

```diff
@@ -1,10 +1,218 @@
-import XCTest
+import ImperialAuth0
+import ImperialDiscord
+import ImperialDropbox
+import ImperialFacebook
+import ImperialGitHub
+import ImperialGitlab
+import ImperialGoogle
+import ImperialKeycloak
+import ImperialMicrosoft
+import Testing
+import XCTVapor
+
 @testable import ImperialCore
 
-class ImperialTests: XCTestCase {
-    func testExists() {}
-    
-    static var allTests: [(String, (ImperialTests) -> () -> ())] = [
-        ("testExists", testExists)
-    ]
+@Suite("Imperial Tests")
+struct ImperialTests {
+    @Test("Auth0 Route")
+    func auth0Route() async throws {
+        try await withApp { app in
+            try await app.test(
+                .GET, "/auth0",
+                afterResponse: { res async throws in
+                    #expect(res.status == .notFound)
+                })
+
+            try app.oAuth(from: Auth0.self, authenticate: "auth0", callback: "auth0-auth-complete", redirect: "/")
+
+            try await app.test(
+                .GET, "/auth0",
+                afterResponse: { res async throws in
+                    #expect(res.status != .notFound)
```

Comment:
Can we do better than not 404

Reply:
Testing this is hard, but I improved it a bit, explaining with a comment why the authentication fails for each service